src/service_daemon.rs

```diff
@@ -770,6 +770,39 @@ struct IntfSock {
     sock: Socket,
 }
 
+/// Enum to represent the IP version.
+#[derive(Debug, Eq, Hash, PartialEq)]
+enum IpVersion {
+    V4,
+    V6,
+}
+
+/// A struct to track multicast notification status for a network interface.
+#[derive(Debug, Eq, Hash, PartialEq)]
+struct MulticastNotificationTracker {
```

Comment:
I think `IfIndexTracker` (or `MulticastSendTracker`) would be easier to understand / map to what it does. It's up to you to decide though.

Reply:
Went with MulticastSendTracker, thx for the suggestions.